presto-main/src/main/java/com/facebook/presto/operator/OrderByOperator.java

```diff
@@ -185,23 +213,72 @@ public Page getOutput()
             return null;
         }
 
-        if (currentPosition >= pageIndex.getPositionCount()) {
+        if (!mergedPages.process() || mergedPages.isFinished()) {
```

Comment:
`!mergedPages.process()` doesn't mean that process finished.

Convert processor to iterator and it should be easier to use in this context.